### ISSUE TITLE
Steps I had to take to get signal-backup-decode to compile on ubuntu 20.04

### ISSUE THREAD

**Issue by pale2hall** [pajowu/signal-backup-decode#33]
On Ubuntu 20.04 I had to do the following to get it to compile:

- Install Rust (obvious, but I haven't played with Rust yet)
- add ~/.cargo/bin to my PATH
- sudo apt install libssl-dev
- export OPENSSL_LIB_DIR="/usr/lib/x86_64-linux-gnu"
- export OPENSSL_INCLUDE_DIR="/usr/include/openssl"
- sudo apt install libsqlite3-dev

**Comment by tmo1** [comment]
And here are the steps I needed on Debian unstable:

- Install Rust: `apt install rustc`
- Install the SQLite and OpenSSL development files: `apt install libsqlite3-dev libssl-dev` [I actually already had the latter installed.]
- Install pkg-config: `apt install pkg-config`

I didn't need the other steps mentioned above (I just ran the binary directly with `~/path/to/signal-backup-decode`).

I suggest that these various dependencies be explicitly documented in the `README`.

But it certainly is an awesome utility - thank you!

**Comment by mutlusun** [comment]
Hello,

do you think the changes in #37 are sufficient?

**Comment by tmo1** [comment]
> Hello,
> 
> do you think the changes in #37 are sufficient?

Thanks! Looks good to me. I'd probably mention `apt install rustc` for Debian in the README as well, but that's your call.
Thanks again for your work on this.

**Comment by mutlusun** [comment]
Thanks for your feedback!

I added a link to the rust installation page. The website prefers installation using rustup and not the debian package. Thus, I would leave the decision to the user.

**Comment by tmo1** [comment]
> Thanks for your feedback!
> 
> I added a link to the rust installation page. The website prefers installation using rustup and not the debian package. Thus, I would leave the decision to the user.

Fair enough (although for some reason, I don't see the link to the rust installation page).

**Comment by ghost** [comment]
I would add:
sudo apt install build-essentials 

I tried it on a new Ubuntu installation and of course it needed a C compiler..
